### PR TITLE
Goodie Template Updates

### PR DESCRIPTION
**Pull request by moollaza**
- Update the Goodie Perl template to suggest structured answers w/ full templating.
    - If we plan to continue supporting the simplified templating, we could use that as the default?
- Added a JS skeleton for a Goodie, adapted from Russell's "Game of Life"
    - This won't be used until the DuckPAN template code is updated to use the file

/cc @zachthompson

### CODE REVIEW

File: template/lib/DDG/Goodie/Example.pm [hunk 1]
```diff
@@ -9,14 +10,33 @@ use strict;
 zci answer_type => '<: $lia_name :>';
 zci is_cached   => 1;
 
+name '<: $lia_name :>';
```
Comment (zachthompson) [inline template/lib/DDG/Goodie/Example.pm]: Not sure I understand why we are re-adding this?

File: template/share/goodie/example/example.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+(function(env) {
+
+    console.log("DDH.'<: $lia_id :>'.build");
+
+    // define private variables and functions here
+
+    DDH.'<: $lia_id :>'.build = function(ops) {
+
+        return {
+            id: "life",
```
Comment (zachthompson) [inline template/share/goodie/example/example.js]: Can this be set by ops.id or does it need to be hardcoded?

File: template/lib/DDG/Goodie/Example.pm [hunk 3]
```diff
@@ -9,14 +10,33 @@ use strict;
...
+
+    # Optional - Regular expression guard
+    # use this approach to ensure the remainder matches a pattern
+    # i.e. it only contains letters, or numbers, or contains certain words
+    #
+    # return unless qr/^\w+/;
+
+    return "plain text response",
+        structured_answer => {
+                id => '<: $lia_id :>', # Should be an existing Instant Answer topic
```
Comment (zachthompson) [inline template/lib/DDG/Goodie/Example.pm]: Comment is not quite right.

File: template/lib/DDG/Goodie/Example.pm [hunk 4]
```diff
@@ -9,14 +10,33 @@ use strict;
...
+                name => 'Answerbar Tab Name', # Should be an existing Instant Answer topic
+                data => { q => $_ },
+                templates => 1
```
Comment (zachthompson) [inline template/lib/DDG/Goodie/Example.pm]: Shouldn't we be doing...?
``` Perl
templates => {
}
```

File: template/lib/DDG/Goodie/Example.pm [hunk 5]
```diff
@@ -9,14 +10,33 @@ use strict;
...
+    # Optional - Regular expression guard
+    # use this approach to ensure the remainder matches a pattern
+    # i.e. it only contains letters, or numbers, or contains certain words
+    #
+    # return unless qr/^\w+/;
+
+    return "plain text response",
+        structured_answer => {
+                id => '<: $lia_id :>', # Should be an existing Instant Answer topic
+                name => 'Answerbar Tab Name', # Should be an existing Instant Answer topic
```
Comment (zachthompson) [inline template/lib/DDG/Goodie/Example.pm]: Comment not right here either

File: template/lib/DDG/Goodie/Example.pm (hunk 1 above)
Comment (moollaza) [inline template/lib/DDG/Goodie/Example.pm]: Russell had it in his code, I figured it was there for a reason, but I'll remove it.

File: template/lib/DDG/Goodie/Example.pm (hunk 4 above)
Comment (moollaza) [inline template/lib/DDG/Goodie/Example.pm]: Crap, just realized I forgot to push my last commit -- I had fixed that!

**Comment by moollaza** [comment]
Looks like all my commits didn't make it to the PR. Sorry, will update :disappointed:

**Comment by moollaza** [comment]
@zachthompson alright, this looks much better now :)

More in-line with our current Spice template. I'll submit some updates to Spice too

File: template/lib/DDG/Goodie/Example.pm [hunk 6]
```diff
@@ -7,16 +8,55 @@ use DDG::Goodie;
...
+
+            templates => {
+                group => "text",
+                # options => {
+                #
+                # }
+            }
+        };
+    };
 };
```
Comment (zachthompson) [inline template/lib/DDG/Goodie/Example.pm]: Extra brace here.

Comment (moollaza) [inline template/lib/DDG/Goodie/Example.pm]: Whoops. Fixed :+1:

**Comment by zachthompson** [comment]
Thank you.

**Comment by srvsh** [comment]
@moollaza the variable `lia_id` is used in [`Example.pm`](https://github.com/duckduckgo/zeroclickinfo-goodies/blob/master/template/lib/DDG/Goodie/Example.pm#L39) and [`example.js`](https://github.com/duckduckgo/zeroclickinfo-goodies/blob/master/template/share/goodie/example/example.js), but there's no corresponding variable in [`New.pm`](https://github.com/duckduckgo/p5-app-duckpan/blob/master/lib/App/DuckPAN/Cmd/New.pm#L63-L69). Would that be an issue?

As of now, on the generated files have an empty string where the `lia_id` is present on my computer.

**Comment by zachthompson** [comment]
That's a bug @srvsh.

**Pull request merged**